bindings/cpu_profiler.cc

```diff
@@ -52,6 +57,171 @@ v8::CpuProfilingLoggingMode GetLoggingMode() {
   return kDefaultLoggingMode;
 }
 
+class SentryProfile;
+
+enum class ProfileStatus {
+  kNotStarted,
+  kStarted,
+  kStopped,
+};
+
+class HeapStatisticsTimer {
+private:
+  uint64_t period_ms;
+  std::unordered_map<std::string, std::function<void(uv_timer_t*, uint64_t, v8::HeapStatistics&)>> listeners;
+  v8::Isolate* isolate;
+  v8::HeapStatistics heap_stats;
+
+public:
+  uv_timer_t timer;
```

Comment:
this shouldn't need to be public. I suggested some changes further down the PR about making `add_listener` and `remove_listener` instance methods instead of static methods, if you did that this could be private